src/utils/breakdown_validation.py

```diff
@@ -0,0 +1,112 @@
+"""Function to validate the breakdown totals from the imputation."""
+import os
+import logging
+import pandas as pd
+import numpy as np
+from pandas import DataFrame as pandasDF
+
+
+BreakdownValidationLogger = logging.getLogger(__name__)
+
+"""Function to validate the breakdown totals from the imputation."""
+def breakdown_validation(df: pd.DataFrame) -> dict:
+    """Function to check that the breakdown values match the criteria provided. 
+    
+    Args:
+        df - the BERD 2023 form dataframe that is to be checked
+    Returns:
+        A boolean dictionary and messages which update based upon if there are any
+        issues with the logic of the columns.   
+    
+    """
+    bool_dict = {}
+    msg = ""
+    
+    df['check1'] = df['222'] + df['223'] != df['203']
```

Comment:
There are a lot of errors being produced because the total value in each check = 0, so the breakdowns are nulls.  Can you add an additional step, so that if the total value = 0, the breakdown totals are set to 0.  That will reduce the unnecessary error messages.